### DISCUSSION TITLE
Frequent Checksum Errors During Scrub on ZFS Pool

### DISCUSSION THREAD

**Discussion by KoffeinKaio** [openzfs/zfs#16445] [Q&A]
Hey everyone,

I'm experiencing an issue where my ZFS pool throws checksum errors during every scrub. Sometimes these errors pop up early in the process, and other times they appear during the last hours of the scrub.
Even had one scrub without any.

Here’s what I’ve tried so far:

* Changed all cables
* changed the psu
* Split some of the drives between the motherboard SATA and a PCI SATA card.
* Installed ECC RAM.

My setup:

8x 20TB Seagate drives (no SMART errors).

Ryzen 3700x

Asrock Phantom Gaming Pro 4

Munchkin unbuffered ECC DDR4 3200 / 64G

lspci sata:
```
02:00.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 51)
03:00.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 51)
```

Controller is passed through via PCI passthrough in a Debian VM (proxmox HV).
Debian 12 with backports.

Kernel: 6.9.12

ZFS version: 2.2.4-2

Here’s the current state of the disks/pool:
```
        NAME                                   STATE     READ WRITE CKSUM
        Data                                   ONLINE       0     0     0
          raidz2-0                             ONLINE       0     0     0
            ata-ST18000NM003D-3DL103_ZVT4WHGT  ONLINE       0     0     1
            ata-ST18000NM003D-3DL103_ZVT5NTBG  ONLINE       0     0     0
            ata-ST18000NM003D-3DL103_ZVT5MER8  ONLINE       0     0     0
            ata-ST18000NM003D-3DL103_ZVT0SCR9  ONLINE       0     0     0
            ata-ST18000NM003D-3DL103_ZVT07MLA  ONLINE       0     0     1
            ata-ST18000NM003D-3DL103_ZVT9JNZR  ONLINE       0     0     1
            ata-ST18000NM003D-3DL103_ZVT9C7NM  ONLINE       0     0     0
            ata-ST18000NM003D-3DL103_ZVTAEL6D  ONLINE       0     0     0
```

Despite these efforts, the checksum errors persist. Sometimes just on one drive, sometimes on 3 and so on.

Does anyone have ideas on what might be causing this issue or suggestions for fixing it? Where could the problem lie?

Thanks in advance for any help!

**Comment by nakinney** [comment]
I've went through something similar were I was getting errors. Replaced cables and PSU in attempt to fix. For me, it was a hot swappable drive case I was using (SilverStone Technology SST-CS380B-USA). When I made direct connections to the drives the errors went away.

**Comment by KoffeinKaio** [comment]
My drives are directly connected to the onboard SATA ports

**Comment by KoffeinKaio** [comment]
update: 
imported the pool directly on the HV without vfio passthrough - chksum errors in the first hour of the scrub
